packages/core/src/app/db/range-bucket.ts

```diff
@@ -282,6 +284,23 @@ export class BaseRangeBucket implements RangeBucket {
     return ranges
   }
 
+  /**
+   * Creates an iterator with some options.
+   * @param options Parameters for the iterator.
+   * @returns the iterator instance.
+   */
+  public iterator(options?: IteratorOptions): RangeIterator {
+    const rangeIterator = new BaseRangeIterator(
+      this.db,
+      {
+        ...options,
+        prefix: this.addPrefix(this.prefix),
```

Comment:
Should this just be `this.prefix`, or do we want the prefix twice?

Reply:
Ah good catch! This should be `options.prefix` not `this.prefix` as it is here -- https://github.com/plasma-group/pigi/blob/master/packages/core/src/app/db/bucket.ts#L80-L85